lib/foreman_maintain/concerns/upstream.rb

```diff
@@ -0,0 +1,109 @@
+module ForemanMaintain
+  module Concerns
+    module Upstream
+      # TO DO - have a special file to keep Foreman and Katello mapping
+      VERSION_MAPPING = {
+        '3.2' => '4.4',
+        '3.3' => '4.5',
+        'nightly' => 'nightly'
+      }.freeze
+
+      def server_url
+        if el?
+          'https://yum.theforeman.org/'
+        elsif debian_or_ubuntu?
+          'https://deb.theforeman.org/'
+        else
+          raise 'Unknown operating system detected!'
+        end
+      end
+
+      def foreman_release_pkg_url(version)
+        if el?
+          "#{server_url}releases/#{version}/#{el_short_name}/x86_64/foreman-release.rpm"
+        elsif debian_or_ubuntu?
+          "#{server_url}pool/#{os_version_codename}/#{version}"\
+          '/f/foreman-release/foreman-release.deb'
```

Comment:
Today, this URL is 404 as we don't have a `latest` symlink. I'm fine merging this PR without it being present, but we need a redmine to track it?

Reply:
here we go: https://projects.theforeman.org/issues/35349